### PR TITLE
docs(python): add "see also" entries to ne/eq_missing and update related examples

### PR DESCRIPTION
**Pull request by alexander-beedie**
Ref: #10657.

### CODE REVIEW

File: py-polars/polars/series/series.py [hunk 1]
```diff
@@ -608,16 +608,42 @@ def eq_missing(self, other: Expr) -> Expr:  # type: ignore[misc]
...
+        [
+            false
+            true
+            true
+        ]
+
+        See Also
+        --------
+        ne_missing
+        eq
```
Comment (stinodego) [inline py-polars/polars/series/series.py:644]: See Also section should be before Examples!

**Comment by stinodego** [comment]
Something got messed up between updating Ruff and PRs not being rebased before merging.

Can you please add `# noqa: PTH207` to the line failing ruff in the CI?

File: py-polars/polars/series/series.py (hunk 1 above)
Comment (alexander-beedie) [reply py-polars/polars/series/series.py:644]: Ok; we're definitely not doing that consistently though. 
Can do a run-through and update elsewhere too.

Comment (stinodego) [reply py-polars/polars/series/series.py:644]: Sure that would be welcome.

The guidelines are here:
https://numpydoc.readthedocs.io/en/latest/format.html#sections

**Comment by alexander-beedie** [comment]
> Something got messed up between updating Ruff and PRs not being rebased before merging.
> 
> Can you please add `# noqa: PTH207` to the line failing ruff in the CI?

No problem; tracked it down to https://github.com/pola-rs/polars/pull/10665; @ritchie46, you might need to `make requirements`? :))

**Comment by ritchie46** [comment]
> No problem; tracked it down to https://github.com/pola-rs/polars/pull/10665; @ritchie46, you might need to make requirements? :))

Whoops! Will do!

**Comment by stinodego** [comment]
The regex master is at it again :smile: feel free to merge if CI passes!

And thanks for the clarification on eq_missing!

**Review by stinodego — APPROVED** [review]

**Pull request merged**